tests/test_infer/test_inference_engine.py

```diff
@@ -0,0 +1,29 @@
+import transformers
+
+from colossalai.inference.core.config import InferenceConfig
+from colossalai.inference.core.engine import InferenceEngine
+
+
+def test_inference_engine():
+    model = transformers.LlamaForCausalLM(
+        transformers.LlamaConfig(
+            vocab_size=20000, hidden_size=512, intermediate_size=1536, num_attention_heads=4, num_hidden_layers=4
+        )
+    )
+    inference_config = InferenceConfig(model, "hf-internal-testing/llama-tokenizer")
```

Comment:
The user will not pass args like they, they will either give one of the below:
1. a path or name of a huggingface repository
2. a model object and a tokenizer object